src/pysonar_scanner/configuration/cli.py

```diff
@@ -20,24 +20,36 @@
 import argparse
 
 from pysonar_scanner.configuration import properties
+from pysonar_scanner.exceptions import UnexpectedCliArgument
 
 
 class CliConfigurationLoader:
 
     @classmethod
     def load(cls) -> dict[str, any]:
-        args = cls.__parse_cli_args()
+        args, unknown_args = cls.__parse_cli_args()
         config = {}
         for prop in properties.PROPERTIES:
             if prop.cli_getter is not None:
                 value = prop.cli_getter(args)
                 config[prop.name] = value
 
+        # Handle unknown args starting with '-D'
+        for arg in unknown_args:
+            if not arg.startswith("-D"):
+                raise UnexpectedCliArgument(f"Unexpected argument: {arg}")
+            key_value = arg[2:].split("=", 1)
+            key, value = key_value
+            config[key] = value
```

Comment:
If an argument is provided without any `=` in it, it will crash here. I think we want to support it because we could have unknown argument in the form `-Dsonar.my.unknowm.flag`. In this case, we could just assign `True` as value. 
```suggestion
            config[key] = True
```
I think it is also an acceptable solution to not support argument without `=` in it, but we should fail nicely on it : 
```python
if len(key_value) != 2 : 
    raise UnexpectedCliArgument(f"Unexpected argument should containt at least one = character : {arg}")
```

Reply:
I think treating args without `=` as flags makes sense, I went for that. Good catch!